torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -671,8 +662,16 @@ def view_avoid_dupes_with_primals(t):
         # This analysis function returns *only* the outputs that are meant to be tangents to the backwards.
         # Anything that aliases (inputs returned in the fw due to metadata mutations, or outputs that alias inputs/intermediates)
         # are *regenerated* later, and not used directly in the autograd graph
+        def _plain_fake_tensor_like_subclass(x):
+            t = torch.empty(x.shape, dtype=x.dtype, device=x.device)
```

Comment:
this is going to allocate actual cuda memory at compile time, which we probably don't want. You probably want this instead:
```
with detect_fake_mode():
    Return torch.empty(x.shape, dtype=x.dtype, device=x.device, layout=x.layout)
```